wire-ios-request-strategy/Sources/Request Strategies/Conversation/ConversationService.swift

```diff
@@ -113,32 +131,60 @@ public final class ConversationService: ConversationServiceInterface {
                 allowServices: allowServices
             ),
             enableReceipts: enableReceipts,
-            messageProtocol: messageProtocol,
-            completion: completion
-        )
+            messageProtocol: messageProtocol) { result in
+                switch result {
+                case .failure(.networkError(.unreachableDomains(let domains))):
+                    retry(excludingDomains: domains)
+
+                default:
+                    completion(result)
+                }
+            }
     }
 
     private func internalCreateGroupConversation(
         name: String?,
         users: Set<ZMUser>,
         completion: @escaping (Swift.Result<ZMConversation, ConversationCreationFailure>) -> Void
     ) {
+        func retry(excludingDomains domains: Set<String>) {
+            let usersToExclude = users.belongingTo(domains: domains)
+            let usersToAdd = Set(users).subtracting(usersToExclude)
+            internalCreateGroupConversation(
+                teamID: nil,
+                name: name,
+                users: usersToAdd,
+                failedToAddUsers: usersToExclude,
+                accessMode: ConversationAccessMode(),
+                accessRoles: [],
+                enableReceipts: false,
+                messageProtocol: .proteus,
+                completion: completion)
+        }
```

Comment:
Instead of having two separate retries (one for team conversation and one for personal), could we have a single one at the inner most method? In other words, leave the `internalCreateTeamGroupConversation` and `internalCreateGroupConversation` methods unchanged, and in `internalCreateGroupConversation` (the one with all the arguments) we can catch the error, perform the retry, and add the system message?